arrow-core/src/main/kotlin/arrow/core/extensions/list/hash/ListKHash.kt

```diff
@@ -0,0 +1,41 @@
+package arrow.core.extensions.list.hash
+
+import arrow.core.extensions.ListKHash
+import arrow.typeclasses.Hash
+import arrow.core.hash as _hash
+import arrow.core.hashWithSalt as _hashWithSalt
+import kotlin.Int
+import kotlin.Suppress
+import kotlin.collections.List
+import kotlin.jvm.JvmName
+
+@JvmName("hash")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("hash(HA)", "arrow.core.hash"))
+fun <A> List<A>.hash(HA: Hash<A>): Int =
+  _hash(HA)
+
+@JvmName("hashWithSalt")
+@Suppress(
+  "UNCHECKED_CAST",
+  "USELESS_CAST",
+  "EXTENSION_SHADOWED_BY_MEMBER",
+  "UNUSED_PARAMETER"
+)
+@Deprecated("@extension projected functions are deprecated", ReplaceWith("hashWithSalt(HA, arg1)", "arrow.core.hashWithSalt"))
+fun <A> List<A>.hashWithSalt(HA: Hash<A>, arg1: Int): Int =
+  _hashWithSalt(HA, arg1)
+
+object List {
```

Comment:
nit, this object may need deprecation too if someone has ever projected other functions on it beside using ours